### PR TITLE
Added Longest Substring with K Distinct Characters in Cpp

### PR DESCRIPTION
**Pull request by bitaashna**
### Have you read the [Contributing Guidelines on Pull Requests]
Yes,I have.

### Description
Added the **'Longest Substring with K Distinct Characters'** program in Cpp, solved using the **'Sliding Window Technique'**

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2304 (Cpp) - mentioned under **'Variable Window Problems'**

### CODE REVIEW

File: C-Plus-Plus/cp/longest_substring_with_k_unique_characters.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        //Updating count of current letter
+        charCount[str[end]]++;
+        //unique elements less than required increase window size
+        if (charCount.size() < k)
+            end++;
+        //if unique element equals to required we need to check
+        //if on increasing the window size whether repetitive
+		// element is present or not
+        //if present then update it as ans
+        //increase window size
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/longest_substring_with_k_unique_characters.cpp]: Use multi-line comments

File: C-Plus-Plus/cp/longest_substring_with_k_unique_characters.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        else if (charCount.size() == k) {
+            ans = max(ans, end - start + 1);
+            end++;
+        }
+        else if (charCount.size() > k) {
+            // If there are more than k unique characters in
+            // current window, remove from left side
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/longest_substring_with_k_unique_characters.cpp]: Use multi-line comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/longest_substring_with_k_unique_characters.cpp (hunk 2 above)
Comment (bitaashna) [reply C-Plus-Plus/cp/longest_substring_with_k_unique_characters.cpp]: done

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -87,12 +87,7 @@
 - [Aggressive cows](cp/aggressive_cows.cpp)
 - [Minimum operations to make sum of each column & row of square matrix equal](cp/matrix_min_operations_for_equal_sum.cpp)
 - [Expressing n as a product of prime numbers](cp/prime_factors.cpp)
-- [Maximum of Minimum for every window size](cp/max_of_min_subarray.cpp)
-- [Maximum Sum Subarray of Size k](cp/max_sum_subarray_of_size_k.cpp)
-- [Maximum of all subarray of size k](cp/maximum_of_all_subarray_of_size_k.cpp)
-- [First Negative Integer in Every k sized window](cp/1st_negative_no_in_every_window_of_size_k.cpp)
-- [Rotating an NxN 2D array in 90 degrees anticlockwise direction](cp/rotate2darray.cpp)
-- [Count occurence of Anagram](cp/count_occurences_of_anagrams.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Why so many removals?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md (hunk 3 above)
Comment (bitaashna) [reply C-Plus-Plus/README.md]: i was facing merge conflict in my main branch beacause of the pr you merged just now so I deleted them from readme.Should I add them again?

Comment (bitaashna) [reply C-Plus-Plus/README.md]: I have changed it.Have a look!

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**